### ISSUE TITLE
Code documentation

### ISSUE THREAD

**Issue by Farmerjoe12** [Progressive-Learning-Platform/plpTool-prototype#89]
Is there any way the project could be documented and maintained in a way that encourages contribution from others? I'm using version 5.2 of PLPT and found my way here to check out the source code, but I can't really understand what's going on. There's no readme or wiki explaining how the tool works, so someone can't easily understand how the code works. If it was better documented it would encourage others to help and work on it. If it's your intention to keep it small that's fine, I'd like to help out if I could.

**Comment by harsha-kadekar** [comment]
Hi thank you for looking into the source and showing interest in the project. Yes we know that project has very poor documentation. In the next 3 months, I will be spending some significant time in documentation as well as developing unit tests for the code. Any helping hands would be greatly appreciated.

The PLPTool-prototype source is PLPTool 6.0, which is still under development. Even older 5.2 source has very limited documentation. I will be working on both of them.

**Comment by tobielf** [comment]
@Farmerjoe12 Yes, we are planning to complete the documentation and refining the source code this summer. I am glad you would like to help. You can check out some commit log on master branch to understand the work we have done.

**Comment by tobielf** [comment]
1. Create a branch from master, name it "refactor-YOURNAME".
2. Add the comment to your designated code file list, follow the java doc best practice.
3. Send out the Pull/Request, I will give out the review feedback based on this stage work.
4. Merge your changes into master.

Let's start from src/edu/asu/plp/tool/backend/isa/events first.
@mjadhav1 AssemblerControlEvent.java and AssemblerResultEvent.java
@prashanthrk92 DeviceInputEvent.java and DeviceOutputEvent.java
@sschava1 MemWatchRequestEvent.java and MemWatchRespondEvent.java
@tusharpandit18 RegWatchRequestEvent.java and RegWatchRespondEvent.java
@tanpuresiddhant SimulatorControlEvent.java and UpdateASMEvent.java

For better understanding the communication between backend and frontend, you can focus on these events while you try to run the PLP tool step by step using an IDE.

Send out your Pull/Request before Thursday, I will do the review and feedback on Friday.
